File: services/game/deploy-lambda.js
const { LambdaClient, UpdateFunctionCodeCommand } = require('@aws-sdk/client-lambda');
const fs = require('fs');
const path = require('path');

const client = new LambdaClient({ region: 'us-east-1' });

async function deployLambda() {
  const zipPath = path.join(__dirname, 'function.zip');
  const zipBuffer = fs.readFileSync(zipPath);

  console.log(`📦 Deploying ${zipBuffer.length} bytes to Lambda...`);

  const command = new UpdateFunctionCodeCommand({
    FunctionName: 'MemoryGame-GameService-dev',
    ZipFile: zipBuffer,
  });

  try {
    const response = await client.send(command);
    console.log('✅ Deployment successful!');
    console.log(`Status: ${response.LastUpdateStatus}`);
    console.log(`Code Size: ${response.CodeSize} bytes`);
  } catch (error) {
    console.error('❌ Deployment failed:', error.message);
    process.exit(1);
  }
}

deployLambda();

File: services/game/deploy-lambda.py
#!/usr/bin/env python3
import boto3
import sys

def deploy_lambda():
    client = boto3.client('lambda', region_name='us-east-1')
    
    with open('function.zip', 'rb') as f:
        zip_data = f.read()
    
    print(f"📦 Deploying {len(zip_data)} bytes to Lambda...")
    
    try:
        response = client.update_function_code(
            FunctionName='MemoryGame-GameService-dev',
            ZipFile=zip_data
        )
        print("✅ Deployment successful!")
        print(f"Status: {response['LastUpdateStatus']}")
        print(f"Code Size: {response['CodeSize']} bytes")
    except Exception as e:
        print(f"❌ Deployment failed: {e}")
        sys.exit(1)

if __name__ == '__main__':
    deploy_lambda()
